git/refs/symbolic.py

```diff
@@ -171,7 +172,14 @@ def _get_ref_info_helper(
         tokens: Union[None, List[str], Tuple[str, str]] = None
         repodir = _git_dir(repo, ref_path)
         try:
-            with open(os.path.join(repodir, str(ref_path)), "rt", encoding="UTF-8") as fp:
```

Comment:
When looking at how `gitoxide` (and `git`) do it I noticed that this could be much simpler.
Ref-names can't be anything they want, they are very limited in what's allowed and what is not (see [this validation code as reference](https://github.com/Byron/gitoxide/blob/dff90531e49a153a38c1cb01c8fdc7ff9bad7e81/gix-validate/src/tag.rs#L48)).

It would be enough to reject any `ref_path` that has a parent-dir component in it.

What do you think?

Reply:
Ah yeah you're right! I fixed it so that now it only checks for `..` in the `ref_path`